src/txn/txn.c

```diff
@@ -1155,6 +1155,98 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
     return (ret);
 }
 
+/*
+ * __txn_commit_timestamps_verbose --
+ *     Print warning messages when encountering surprise timestamp usage.
+ */
+static inline int
+__txn_commit_timestamps_verbose(WT_SESSION_IMPL *session)
+{
+#ifndef HAVE_DIAGNOSTIC
```

Comment:
We will want to change this, so that the diagnostic build just does a bit more checking.